lstchain/io/lstcontainers.py

```diff
@@ -95,7 +95,7 @@ class DL1ParametersContainer(Container):
     mc_core_x = Field(None, 'Simulated impact point x position', unit=u.m)
     mc_core_y = Field(None, 'Simulated impact point y position', unit=u.m)
     mc_h_first_int = Field(None, 'Simulated first interaction height', unit=u.m)
-    mc_type = Field(-1, 'Simulated particle type')
+    mc_type = Field(np.nan, 'Simulated particle type')
```

Comment:
same here, -1 is ok, particle id is also a positive integer.

Reply:
> > I did not foresee the writing type issue... (these parameters are requested to be integers and `nan` is a float)
> > I will try and think about an elegant solution but suggestions are welcome.
> 
> Then the default necessarily has to be an integer outside of the physical range, right? And filter those before passing the events to the training (and to the application of the trained models)

Yes that was basically how it was implemented.
`nan` has the advantage to always be outside the physical range and does not need extra care.
But yes, with the type issue, I will introduce some more precise filtering...

Reply:
actually I think that in the corsika nomenclatura, negative integers are possible and meant for anti-particles (-1 would be positrons then)
maybe I'll rather go for -9999 just in case ;-)

Reply:
No, corsika uses positive numbers:

From the user's guide:
![Screenshot_20210210_212528](https://user-images.githubusercontent.com/5488440/107567849-869cb880-6be6-11eb-8fb4-5e9fe9bf78d7.png)

Reply:
:o 
thanks!
I would have sworn I had seen negative values in the past, my mistake!

Reply:
Sorry, triple checking this, I found that simtel does not use the CORSIKA codes anymore.

From `io_hess.h`:

```
   int primary_id;      ///< Particle ID of primary. Was in CORSIKA convention
                        ///< where detector_prog_vers in MC run header was 0,
                        ///< and is now 0 (gamma), 1(e-), 2(mu-), 100*A+Z
                        ///< for nucleons and nuclei, negative for antimatter.
```

Reply:
So this should be 0 for gamma, 101 for proton.

Reply:
Yes, verified for our current simulations. 0 = gamma, 101 = proton, negative numbers possible.

Reply:
Thanks for the counter check!
I'll update accordingly.